packages/evolution-frontend/src/components/inputs/InputRadioNumber.tsx

```diff
@@ -0,0 +1,159 @@
+import { IconProp } from '@fortawesome/fontawesome-svg-core';
+import { CliUser } from "chaire-lib-common/lib/services/user/userType";
+import { UserInterviewAttributes } from "evolution-common/lib/services/interviews/interview";
+import {InputRadioNumberType} from "evolution-common/lib/services/widgets";
+
+type InputRadioNumberProps<CustomSurvey, CustomHousehold, CustomHome, CustomPerson> = {
+    id: string;
+    inputRef?: React.LegacyRef<HTMLTextAreaElement>;
+    onValueChange: (e: any, customValue?: string) => void;
+    widgetConfig: InputRadioNumberType;
+    size?: 'small' | 'medium' | 'large';
+    icon?: IconProp;
+    user: CliUser;
+    interview: UserInterviewAttributes<CustomSurvey, CustomHousehold, CustomHome, CustomPerson>;
+    inputIconPath?: { iconPath: string; iconSize: string; };
+}
+
+type InputRadioChoiceProps = {
+    id: string,
+    value: number;
+    icon?: IconProp;
+    inputIconPath?: { iconPath: string; iconSize: string; };
+    onChange: ChangeEventHandler<HTMLInputElement>;
+}
+
+const InputRadioNumberChoice = (
+    {
+        id,
+        value,
+        icon,
+        inputIconPath,
+        onChange,
+    }: InputRadioChoiceProps) => {
+    const inputRadioRef: React.RefObject<HTMLInputElement> = React.createRef();
+    return (
+        <div
+            className={'survey-question__input-radio-input-container'}
+        >
+            <div className="label-input-container">
+                <input
+                    type="radio"
+                    id={id}
+                    className={'input-radio'}
+                    value={value}
+                    onChange={onChange}
+                    ref={inputRadioRef}
+                />
+                <label htmlFor={id}>
+                    {icon && (
+                        <span>
+                            <FontAwesomeIcon
+                                icon={icon}
+                                className="faIconLeft"
+                                style={{margin: 'auto 0'}}
+                            />
+                        </span>
+                    )}
+                    {inputIconPath && (
+                        <span>
+                            <img
+                                src={inputIconPath.iconPath}
+                                className="faIconLeft"
+                                style={{height: inputIconPath.iconSize, margin: '0 0.25em 0 0'}}
+                                alt={`${value}`}
+                            />
+                        </span>
+                    )}
+                    <span>{value}</span>
+                </label>
+            </div>
+        </div>
+    );
+};
+
+export const InputRadioNumber = (
+    {
+        id,
+        inputRef,
+        onValueChange,
+        widgetConfig,
+        size,
+        icon,
+        user,
+        interview,
+        inputIconPath
+    }: InputRadioNumberProps<any, any, any, any>) => {
+
+    const [value, setValue] = useState(-1);
```

Comment:
This could be initialized with a prop, but it would mean that a default choice is selected. We would still need to initialize this value to -1 if there is no choice selected by default